Fix downloading packages for updates with multiple builds

AdamWill opened the pull request:
I inadvertently broke this in #3148 - sorry. I tried to make
the handling of the args list cleaner in that commit, but
overlooked that initializing it outside of the `for build in`
loop meant it'd just keep getting appended to, it doesn't get
recreated for each build - so by the time you get to the second
build in an update, you're trying to do something like:

koji download-build foo-1.0-1 bar-1.0-1

which isn't allowed (that's why we run one command per build in
the first place, life would be easier if Koji let us pass it
multiple NVRs...)

Fix this simply by (re)initializing `args` inside the loop, not
outside it.

Signed-off-by: Adam Williamson <awilliam@redhat.com>

AdamWill committed:
Fix downloading packages for updates with multiple builds

I inadvertently broke this in #3148 - sorry. I tried to make
the handling of the args list cleaner in that commit, but
overlooked that initializing it outside of the `for build in`
loop meant it'd just keep getting appended to, it doesn't get
recreated for each build - so by the time you get to the second
build in an update, you're trying to do something like:

koji download-build foo-1.0-1 bar-1.0-1

which isn't allowed (that's why we run one command per build in
the first place, life would be easier if Koji let us pass it
multiple NVRs...)

Fix this simply by (re)initializing `args` inside the loop, not
outside it.

Signed-off-by: Adam Williamson <awilliam@redhat.com>
Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp commented:
The patch looks good to me and I verified that it works, but I'll add a test covering this case before we get this merged.

AdamWill commented:
Yeah, sorry I didn't do that, I needed to fix it in a hurry because the problem was making openQA's tests invalid.

Another thing I think it should probably do is exit non-0 if any of the downloads fails (but it should try them all before doing so).

nphilipp committed:
Verify the correct number of received items.

Previously, if the user specified builds to download, this was compared
to the number of returned updates. This number happened to be identical
if only one build was requested, or only builds from different updates.

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp commented:
I've pushed the test and a fix needed for the test to work, I'll wait for the CI to process.

> Another thing I think it should probably do is exit non-0 if any of the downloads fails (but it should try them all before doing so).

Let's do this in a separate issue/PR. There's also the issue of that it can download too many builds e.g. if you specify one from a multi-build update.

AdamWill commented:
"There's also the issue of that it can download too many builds e.g. if you specify one from a multi-build update."

mmhmhhhhh...i'm tempted to say that's intended behaviour. If you only want one build...just use `koji download-build` directly. I've certainly run `bodhi updates download <NEVR>` to download *all the builds from the update containing that build*, just because I happen to know the NEVR and don't want to bother looking up the update ID. I wouldn't say it's valid to assume that just because someone passed a build NEVR, they *only* want to download that build.

nphilipp commented:
> "There's also the issue of that it can download too many builds e.g. if you specify one from a multi-build update."
> 
> mmhmhhhhh...i'm tempted to say that's intended behaviour. If you only want one build...just use `koji download-build` directly. I've certainly run `bodhi updates download <NEVR>` to download _all the builds from the update containing that build_, just because I happen to know the NEVR and don't want to bother looking up the update ID. I wouldn't say it's valid to assume that just because someone passed a build NEVR, they _only_ want to download that build.

Yeah, you're right:

```
Usage: bodhi updates download [OPTIONS]

  Download the builds in one or more updates.
...
  --builds TEXT    Download update(s) by build NVR(s) (comma-separated list)
```
